### ISSUE TITLE
`orderBy`: Make `nulls: 'first' | 'last'` available on non-nullable fields

### ISSUE THREAD

**Issue by Weakky** [prisma/prisma#19760]
## Overview
I'd like to request the behavior of generated types to change a little if possible.  If I have a model such as
```
model Table {
   value: Decimal @db.Money
   otherValue: Decimal?
}
```
And I set the order by in a findMany for example as
```
...
prisma.table.findMany({
   orderBy: {
      value: { sort 'desc', nulls: 'last' }
   }
})
```
This currently results in an error whereas if I sort by `otherValue` it does not.  The reason being that the generated type for the find many args of `value` is simply `Prisma.SortOrder` whereas for `otherValue` it is `Prisma.SortOrderInput | Prisma.SortOrder`.  This is frustrating in my particular case because I'm using postgres, and I can simply pass the nulls argument directly to a non-nullable field, and things just work (I realize there's a no-op here):
```
select *
from Table
order by value asc
null first
```
It's nice to be able to pass this as a no op for the purposes of developing an api against a dataset using this feature.  I want that api to have null order parameters where I can use this feature, but I now have to ensure that I'm only using this feature for specific fields and not others which results in a far greater code footprint, and of course as the schema evolves it will require changes to this larger code footprint, making my use of this feature kind of brittle.

**tldr;** can you have prisma generate `Prisma.SortOrderInput | Prisma.SortOrder` types for sortable field arguments regardless of whether the field is nullable or not?

Just some food for thought - not sure if it's possible or an easy change given you need to accommodate more rdbms than just postgres.

_Originally posted by @ahammouda in https://github.com/prisma/prisma/issues/14377#issuecomment-1409151459_

**Comment by jkomyno** [comment]
Hi there,
As part of our effort to clean up our GitHub repo, we’re closing this issue to help focus on features and improvements that align with our focus areas and those of our community. 

This issue was originally created for internal tracking purposes, but if you feel this is important to you, feel free to reopen the issue and get like-minded users to upvote as we will prioritize fixing issues based on community interest.

For more insight into our priorities and how we’re thinking about the future of Prisma, please see our blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and support!
